### PR TITLE
Task 957 modal using grid

### PR DESCRIPTION
**Pull request by dancingolav**

### CODE REVIEW

File: jdi-light-bootstrap/src/main/java/com/epam/jdi/light/ui/bootstrap/elements/complex/Modal.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.epam.jdi.light.ui.bootstrap.elements.complex;
```
Comment (AUlasen) [inline jdi-light-bootstrap/src/main/java/com/epam/jdi/light/ui/bootstrap/elements/complex/Modal.java]: Move to 
com.epam.jdi.light.ui.bootstrap.elements.composite

File: jdi-light-bootstrap/src/main/java/com/epam/jdi/light/ui/bootstrap/elements/complex/Modal.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import com.epam.jdi.light.elements.pageobjects.annotations.locators.UI;
+import com.epam.jdi.light.ui.bootstrap.elements.common.Button;
+import com.epam.jdi.light.ui.bootstrap.elements.common.Text;
+
+public class Modal extends Section {
+    @UI(".modal-header .modal-title")
+    public Text title;
+    @UI(".modal-header button")
+    public Button closeX;
+}
```
Comment (AUlasen) [inline jdi-light-bootstrap/src/main/java/com/epam/jdi/light/ui/bootstrap/elements/complex/Modal.java]: Some suggestions from @RomanIovlev:

```
public class Modal extends Section {
    @UI(".modal-header .modal-title")
    public Text title;

    public void close() {
        linked(".modal-header button", "Close").click();
    }
}
```

File: jdi-light-bootstrap/src/main/java/com/epam/jdi/light/ui/bootstrap/elements/complex/Modal.java (hunk 1 above)
Comment (dancingolav) [reply jdi-light-bootstrap/src/main/java/com/epam/jdi/light/ui/bootstrap/elements/complex/Modal.java]: For now we have .epam.jdi.light.ui.bootstrap.elements.common and .epam.jdi.light.ui.bootstrap.elements.complex. Should I create new one? complex in jdi-light-bootstrap? (now we have composite in jdi-light-bootstrap-tests only)

Comment (AUlasen) [reply jdi-light-bootstrap/src/main/java/com/epam/jdi/light/ui/bootstrap/elements/complex/Modal.java]: Yes, you should create new package in jdi-light-bootstrap:
com.epam.jdi.light.ui.bootstrap.elements.composite

File: jdi-light-bootstrap-tests/src/main/java/io/github/com/sections/modal/gridmodal/GridModalWindow.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    @UI(".modal-footer")
+    private ModalFooterSaveClose modalFooter;
+
+    public GridModalBody getBody() {
+        return gridModal;
+    }
+
+    public ModalFooterSaveClose getFooter() {
+        return modalFooter;
+    };
```
[the remark this answers is not in the document]
Comment (dancingolav) [reply jdi-light-bootstrap-tests/src/main/java/io/github/com/sections/modal/gridmodal/GridModalWindow.java]: ok

File: jdi-light-bootstrap/src/main/java/com/epam/jdi/light/ui/bootstrap/elements/complex/Modal.java (hunk 2 above)
Comment (dancingolav) [reply jdi-light-bootstrap/src/main/java/com/epam/jdi/light/ui/bootstrap/elements/complex/Modal.java]: Stop. Button closeX has to be removed also. Jst a moment

Comment (dancingolav) [reply jdi-light-bootstrap/src/main/java/com/epam/jdi/light/ui/bootstrap/elements/complex/Modal.java]: button closeX removed

File: jdi-light-bootstrap-tests/src/test/java/io/github/epam/bootstrap/tests/composite/section/modal/GridModalTests.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        cell.is().core().text(textExpected)
+                .and()
+                .css("max-width", startsWith(max_width));
+        cell.unhighlight();
+    }
+
+    @Test
+    public void checkCloseModalButton() {
+        gridModalSection.getGridModalWindow().getBtnClose().highlight("red");
+        gridModalSection.getGridModalWindow().clickBtnClose();
```
Comment (AUlasen) [inline jdi-light-bootstrap-tests/src/test/java/io/github/epam/bootstrap/tests/composite/section/modal/GridModalTests.java]: You should use method close() of Modal object instead.

**Comment by AUlasen** [comment]
Also resolve conflicts, please.

File: jdi-light-bootstrap-tests/src/test/java/io/github/epam/bootstrap/tests/composite/section/modal/GridModalTests.java (hunk 4 above)
Comment (dancingolav) [reply jdi-light-bootstrap-tests/src/test/java/io/github/epam/bootstrap/tests/composite/section/modal/GridModalTests.java]: ok

**Pull request merged**